### PR TITLE
CO-2241 Ajouter un mécanisme pour suivre les parrainages en attente de mandat

### PR DESCRIPTION
**Pull request by jo-bloiz-ornot**
add a field mandate_date for checking if you wait for more than 3 month in mandate state

### CODE REVIEW

**Comment by jo-bloiz-ornot** [comment]
maybe 3 month is too long ?

**Comment by ecino** [comment]
> maybe 3 month is too long ?

Yes it's too long. 3 weeks is better.

File: sponsorship_switzerland/views/res_partner_view.xml [hunk 1]
```diff
@@ -10,4 +10,15 @@
             </field>
         </field>
     </record>
+
+    <record id="action_view_partner_waiting_mandate" model="ir.actions.server" >
```
Comment (ecino) [inline sponsorship_switzerland/views/res_partner_view.xml]: It's better to define a window action, with that you will not even need any method in recurring.contract object!
```suggestion
    <record id="action_view_partner_waiting_mandate" model="ir.actions.act_window">
        <field name="name">Waiting mandate</field>
        <field name="res_model">recurring.contract</field>
        <field name="view_type">form</field>
        <field name="view_mode">tree,form</field>
        <field name="domain" eval="[('state', '=', 'mandate'), ('mandate_date', '<', (DateTime.now() - timedelta(days=25)).strftime('%Y-%m-%d'))]"/>
    </record>
```

**Review by ecino — CHANGES_REQUESTED** [review]
One flake8 issue in migration:
`sponsorship_switzerland/migrations/10.0.1.4.0/post-migration.py:20:31: W291 trailing whitespace`

File: sponsorship_switzerland/views/res_partner_view.xml (hunk 1 above)
Comment (jo-bloiz-ornot) [reply sponsorship_switzerland/views/res_partner_view.xml]: that would take me a loong time to find, thank you

**Review by ecino — APPROVED** [review]

**Pull request merged**